feature/experimental/gribi/otg_tests/fib_failed_due_to_hw_res_exhaust_test/fib_failed_due_to_hw_res_exhaust_test.go

```diff
@@ -0,0 +1,479 @@
+		SetPrefix(advertisedRoutesv4MaskLen).
+		SetCount(vendorSpecRoutecount[args.dut.Vendor()]).SetStep(2)
+
+	args.otg.PushConfig(t, args.otgConfig)
+	time.Sleep(30 * time.Second)
+	args.otg.StartProtocols(t)
+	time.Sleep(30 * time.Second)
+}
+
+// awaitTimeout calls a fluent client Await, adding a timeout to the context.
+func awaitTimeout(ctx context.Context, c *fluent.GRIBIClient, t testing.TB, timeout time.Duration) error {
+	t.Helper()
+	subctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	return c.Await(subctx, t)
+}
+
+// createIPv4Entries creates IPv4 Entries given the totalCount and starting prefix.
+func createIPv4Entries(t *testing.T, startIP string) []string {
+	t.Helper()
+	_, netCIDR, err := net.ParseCIDR(startIP)
+	if err != nil {
+		t.Fatalf("Failed to parse prefix: %v", err)
+	}
+	netMask := binary.BigEndian.Uint32(netCIDR.Mask)
+	firstIP := binary.BigEndian.Uint32(netCIDR.IP)
+	lastIP := (firstIP & netMask) | (netMask ^ 0xffffffff)
+	var entries []string
+	for i := firstIP; i <= lastIP; i++ {
+		ip := make(net.IP, 4)
+		binary.BigEndian.PutUint32(ip, i)
+		entries = append(entries, fmt.Sprint(ip))
+	}
+	return entries
+}
+
+// injectEntry programs gRIBI nh, nhg and ipv4 entry.
+func injectEntry(ctx context.Context, t *testing.T, args *testArgs) {
+	t.Helper()
+	dstIPList := createIPv4Entries(t, fmt.Sprintf("%s/%d", dstIPBlock, 20))
+	vipList := createIPv4Entries(t, fmt.Sprintf("%s/%d", vipBlock, 20))
+	j := uint64(0)
+
+routeAddLoop:
+	for i := uint64(1); i <= uint64(1500); i += 2 {
+		vipNhIndex := i
+		dstNhIndex := vipNhIndex + 1
+
+		args.client.Modify().AddEntry(t,
+			fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(args.dut)).
+				WithIndex(vipNhIndex).WithIPAddress(atePort2.IPv4),
+			fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(args.dut)).
+				WithID(vipNhIndex).AddNextHop(vipNhIndex, 1),
+			fluent.IPv4Entry().WithNetworkInstance(deviations.DefaultNetworkInstance(args.dut)).
+				WithPrefix(vipList[j]).WithNextHopGroup(vipNhIndex),
+		)
+		args.client.Modify().AddEntry(t,
+			fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(args.dut)).
+				WithIndex(dstNhIndex).WithIPAddress(vipList[j]),
+			fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(args.dut)).
+				WithID(dstNhIndex).AddNextHop(dstNhIndex, 1),
+			fluent.IPv4Entry().WithNetworkInstance(deviations.DefaultNetworkInstance(args.dut)).
+				WithPrefix(dstIPList[j]).WithNextHopGroup(dstNhIndex),
+		)
+
+		if err := awaitTimeout(args.ctx, args.client, t, time.Minute); err != nil {
+			t.Logf("Could not program entries via client, got err, check error codes: %v", err)
+		}
+
+		for _, v := range args.client.Results(t) {
+			if v.ProgrammingResult == aftspb.AFTResult_FIB_FAILED {
+				t.Logf("FIB FAILED received %v", v.Details)
+				fibFailedDstRoute = dstIPList[j]
+				break routeAddLoop
+			}
+		}
+		j = j + 1
+		if i >= 1458 {
```

Comment:
What's the logic behind this value (`1458`)? Could you please add a comment to explain?

Reply:
We are filling FIB with BGP Routes and after FIB is full we are trying to program gRIBI routes. Hence taken 1500 unique routes to test if we receive FIB FAILED while programming gRIBI routes.